### DISCUSSION TITLE
Salt and pepper artifact in QSM

### DISCUSSION THREAD

**Discussion by JYProjs** [kschan0214/sepia#41] [Q&A] [answered]
Hi, I am trying to use sepia for our study and am running into an issue where there is salt and pepper artifact in the final QSM map. What leads to the artifact appears somewhere at or before phase unwrapping (e.g echo combining, phase unwrapping, or possibly during the data collection steps) and occurs with multiple toolboxes and on multiple subjects we have scanned. It appears at the boundary where phase wrapping occurs. We use 3T MRI and a multi-echo acquisition. I wanted to reach out to see if a better QSM could be generated with our data and to try and find out what may be causing the artifact. I have attached a few screenshots of the final output and intermediate outputs.

### part-phase_unwrapped:
![Pasted Graphic 3](https://user-images.githubusercontent.com/73326340/190461304-0c76bb02-ff64-4c50-8493-a8576b6494fe.png)![Pasted Graphic 4](https://user-images.githubusercontent.com/73326340/190461325-0c8a8f7a-7ea2-4630-8387-edc23bed6a3e.png)

### field_map
![Pasted Graphic 6](https://user-images.githubusercontent.com/73326340/190461459-6ff1e1f6-7d06-4ca5-a25c-d0bc7829939f.png) ![Pasted Graphic 7](https://user-images.githubusercontent.com/73326340/190461484-03f06875-97b3-4230-a039-3f83575d15ea.png)

### local_field
![Pasted Graphic 9](https://user-images.githubusercontent.com/73326340/190492567-f1fe9936-44ab-4291-802d-713be41dbd2b.png)![Pasted Graphic 10](https://user-images.githubusercontent.com/73326340/190492584-b7325572-f0c4-4320-b91b-ea075d01ec32.png)

### QSM
![Pasted Graphic](https://user-images.githubusercontent.com/73326340/190461654-d4b01304-e982-42ad-b1bc-94ea0da14040.png)![Pasted Graphic 1](https://user-images.githubusercontent.com/73326340/190461619-62465e05-64a7-42c7-aaa8-3a7577b8efe4.png)

Please let me know if I can provide any additional information.

**Comment by kschan0214** [comment]
Hi James,

It seems also to me that the artefact appears in the total field estimation process. The artefact seems to be caused by low SNR in certain brain regions after echo combination/total field estimation and subsequently propagates to the downstream processing steps. 

To have a better insight of the artefact origin, it would be helpful if you can also post the images of the original multi-echo wrapped phase and magnitude data (e.g. same slices for each echo) here, so that I can have a look into data quality. Can you also let me know which echo combination method and phase unwrapping method used?

**Reply by JYProjs** [reply]
Hi Kwok-Shing,

For these images, Optimum weights and Laplacian (MEDI) was used. We have also tried with SEGUE for phase unwrapping and ROMEO for both echo combination and phase unwrapping with similar results. Below are slices of the magnitude and phase data for each echo. We would be happy to provide the nifti files to see if a better QSM could be generated with our data. 

Magnitude:
![Pasted Graphic 43](https://user-images.githubusercontent.com/73326340/190714091-ff7bacaa-6162-469e-9e17-969914cd7515.png)
phase:
![Pasted Graphic 44](https://user-images.githubusercontent.com/73326340/190714075-6ddb66ad-3385-48f4-b561-6328bb89ee5d.png)

**Comment by kschan0214** [comment, marked as answer]
Thanks for sharing the images!

The low SNR issue is caused by the so-called "phase singularity (or open-ended fringe line) in the phase data (as indicated by the arrows). This is the result of sub-optimal coil combination on the phase data, where the differences in phase offset of each coil image cancel each other (destructive interference) in the reconstructed data when the multi-coil data are combined. If by any chance, the raw k-space data or multi-channel images are still available then there might be a way to correct it with the acquired data. However, I'm not aware of any software packages available to correct this issue with the coil combined data (since the offsets are coil-dependent and this information is lost in the reconstructed data). 

Would you mind also letting me know which scanner and software version the data were acquired? I think most of the vendors support combining phase information from the RF array coils appropriately. I might be able to provide more information about how the phase images can be exported correctly for future data acquisition.

![190714075-6ddb66ad-3385-48f4-b561-6328bb89ee5d](https://user-images.githubusercontent.com/39571502/190737427-811149da-0cb1-4c05-80be-f210e16a5d7a.png)

**Reply by JYProjs** [reply]
Thanks so much for your help! We are checking to see if we have the k-space or multi-channel data. If we have the original data, what would you advise to avoid it from appearing from the coil combination? 

We are using the Siemens Prisma model with software version syngo MR E11.

**Reply by kschan0214** [reply]
If the multi-channel data is still available then it may be possible to use ROMEO directly for proper coil combination and phase unwrapping (see https://github.com/korbinian90/ROMEO). You can then use the unwrapped field map of ROMEO in SEPIA. 

For E11, in System-> Miscellaneous->Coil Combination Mode there should be an option called "Adaptive Combine". Make sure the Reconstruction in Contrast->Dynamic is set to Magn./Phase. It is also recommended to check the Prescan Normalize box in Resolution->Filter Image.

**Reply by JYProjs** [reply]
Unfortunately, the data was not stored and looking at the settings, and it seems like we selected "Sum of Squares" instead of "Adaptive Combine" for the combination mode.

**Reply by kschan0214** [reply]
In this case I'm afraid there is not much we could do on the acquired data. I hope that the information could at least be useful for the future data acquisition.

**Comment by jprfm** [comment]
Just an additional point for future acquisitions, in the VE version and if
using the product gradient echo sequence, you can perform acceleration in
only one dimension. If using the vibe sequence you accelerate in multiple
dimensions and will not get these artefacts.
